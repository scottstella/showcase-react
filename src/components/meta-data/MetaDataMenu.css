.meta-data-menu {
  display: flex;
  gap: 30px;
  cursor: pointer;
}

.meta-data-menu__selected {
  font-weight: bold;
}

hr {
  margin-bottom: 30px;
}
