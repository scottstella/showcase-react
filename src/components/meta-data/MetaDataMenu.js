import React, { useState } from "react";
import "./MetaDataMenu.css";

export default function MetaDataMenu(props) {
  const [menuSelection, setMenuSelection] = useState("");

  const makeMenuSelection = (e) => {
    setMenuSelection(e.target.id);
    props.onSelectMetaData(e.target.id);
  };

  return (
    <div>
      <div className="meta-data-menu">
        <div
          id="sets"
          onClick={makeMenuSelection}
          class={`${
            menuSelection === "sets" ? "meta-data-menu__selected" : ""
          }`}
        >
          Sets
        </div>
        <div
          id="classes"
          onClick={makeMenuSelection}
          class={`${
            menuSelection === "classes" ? "meta-data-menu__selected" : ""
          }`}
        >
          Classes
        </div>
        <div
          id="tribes"
          onClick={makeMenuSelection}
          class={`${
            menuSelection === "tribes" ? "meta-data-menu__selected" : ""
          }`}
        >
          Tribes
        </div>
      </div>
      <hr />
    </div>
  );
}
