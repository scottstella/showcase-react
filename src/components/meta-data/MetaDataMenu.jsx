import React, { useState } from "react";
import "./MetaDataMenu.css";

export default function MetaDataMenu(props) {
  const [menuSelection, setMenuSelection] = useState("");

  const makeMenuSelection = (e) => {
    setMenuSelection(e.target.id);
    props.onSelectMetaData(e.target.id);
  };

  return (
    <div>
      <div className="meta-data-menu">
        <div
          id="sets"
          onClick={makeMenuSelection}
          className={`${
            menuSelection === "sets" ? "meta-data-menu__selected" : ""
          }`}
        >
          Sets
        </div>
        <div
          id="classes"
          onClick={makeMenuSelection}
          className={`${
            menuSelection === "classes" ? "meta-data-menu__selected" : ""
          }`}
        >
          Hero Classes
        </div>
        <div
          id="tribes"
          onClick={makeMenuSelection}
          className={`${
            menuSelection === "tribes" ? "meta-data-menu__selected" : ""
          }`}
        >
          Tribes
        </div>
      </div>
      <hr />
    </div>
  );
}
