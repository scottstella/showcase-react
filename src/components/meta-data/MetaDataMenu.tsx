import React, { useState } from "react";
import "./MetaDataMenu.css";

interface MetaDataMenuProps {
  onSelectMetaData: (id: string) => void;
}

const MetaDataMenu: React.FC<MetaDataMenuProps> = ({
  onSelectMetaData,
}: MetaDataMenuProps) => {
  const [menuSelection, setMenuSelection] = useState<string>("");

  const makeMenuSelection = (e: React.MouseEvent<HTMLDivElement>) => {
    const id = e.currentTarget.id;
    setMenuSelection(id);
    onSelectMetaData(id);
  };

  return (
    <div>
      <div className="meta-data-menu">
        <div
          id="sets"
          onClick={makeMenuSelection}
          className={`${
            menuSelection === "sets" ? "meta-data-menu__selected" : ""
          }`}
        >
          Sets
        </div>
        <div
          id="classes"
          onClick={makeMenuSelection}
          className={`${
            menuSelection === "classes" ? "meta-data-menu__selected" : ""
          }`}
        >
          Hero Classes
        </div>
        <div
          id="tribes"
          onClick={makeMenuSelection}
          className={`${
            menuSelection === "tribes" ? "meta-data-menu__selected" : ""
          }`}
        >
          Tribes
        </div>
      </div>
      <hr />
    </div>
  );
};

export default MetaDataMenu;
