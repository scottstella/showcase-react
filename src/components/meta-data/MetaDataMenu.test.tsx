import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import MetaDataMenu from "./MetaDataMenu";

describe("MetaDataMenu", () => {
  const mockOnSelectMetaData = vi.fn();

  beforeEach(() => {
    mockOnSelectMetaData.mockClear();
  });

  it("renders without crashing", () => {
    render(<MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />);
    expect(screen.getByText("Sets")).toBeInTheDocument();
    expect(screen.getByText("Hero Classes")).toBeInTheDocument();
    expect(screen.getByText("Tribes")).toBeInTheDocument();
  });

  it("renders all menu items without initial selection", () => {
    render(<MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />);

    const sets = screen.getByText("Sets");
    const classes = screen.getByText("Hero Classes");
    const tribes = screen.getByText("Tribes");

    expect(sets).not.toHaveClass("meta-data-menu__selected");
    expect(classes).not.toHaveClass("meta-data-menu__selected");
    expect(tribes).not.toHaveClass("meta-data-menu__selected");
  });

  it("applies selected class when menu item is clicked", () => {
    render(<MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />);

    const sets = screen.getByText("Sets");
    fireEvent.click(sets);

    expect(sets).toHaveClass("meta-data-menu__selected");
    expect(screen.getByText("Hero Classes")).not.toHaveClass(
      "meta-data-menu__selected",
    );
    expect(screen.getByText("Tribes")).not.toHaveClass(
      "meta-data-menu__selected",
    );
  });

  it("calls onSelectMetaData with correct id when menu item is clicked", () => {
    render(<MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />);

    fireEvent.click(screen.getByText("Sets"));
    expect(mockOnSelectMetaData).toHaveBeenCalledWith("sets");

    fireEvent.click(screen.getByText("Hero Classes"));
    expect(mockOnSelectMetaData).toHaveBeenCalledWith("classes");

    fireEvent.click(screen.getByText("Tribes"));
    expect(mockOnSelectMetaData).toHaveBeenCalledWith("tribes");
  });

  it("updates selection when different menu item is clicked", () => {
    render(<MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />);

    // Click Sets first
    fireEvent.click(screen.getByText("Sets"));
    expect(screen.getByText("Sets")).toHaveClass("meta-data-menu__selected");

    // Then click Hero Classes
    fireEvent.click(screen.getByText("Hero Classes"));
    expect(screen.getByText("Sets")).not.toHaveClass(
      "meta-data-menu__selected",
    );
    expect(screen.getByText("Hero Classes")).toHaveClass(
      "meta-data-menu__selected",
    );
  });

  it("renders with correct styling", () => {
    const { container } = render(
      <MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />,
    );

    const menu = container.querySelector(".meta-data-menu");
    expect(menu).toBeInTheDocument();

    // Check that horizontal rule is present
    const hr = container.querySelector("hr");
    expect(hr).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <MetaDataMenu onSelectMetaData={mockOnSelectMetaData} />,
    );
    expect(container).toMatchSnapshot();
  });
});
