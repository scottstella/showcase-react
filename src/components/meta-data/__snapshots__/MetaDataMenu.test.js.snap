// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MetaDataMenu > matches snapshot 1`] = `
<div>
  <div>
    <div
      class="meta-data-menu"
    >
      <div
        class=""
        id="sets"
      >
        Sets
      </div>
      <div
        class=""
        id="classes"
      >
        Hero Classes
      </div>
      <div
        class=""
        id="tribes"
      >
        Tribes
      </div>
    </div>
    <hr />
  </div>
</div>
`;
